pkg/ring/tokens.go

```diff
@@ -22,7 +22,7 @@ func (t Tokens) StoreToFile(tokenFilePath string) error {
 
 	// If any operations failed further in the function, we keep the temporary
 	// file hanging around for debugging.
-	f, err := ioutil.TempFile(os.TempDir(), "tokens")
+	f, err := os.Create(tokenFilePath + ".tmp")
```

Comment:
Have you considered the following?

```
ioutil.TempFile(basedir-of-tokenFilePath, "tokens")
```

It would just change the directory while keeping the temp file creation logic.

Reply:
Yes. But I prefer a fixed tmp file in this case to avoid leaving multiple temporary files behind if there was an error (and difficult to tell which was the last temporary file).